src/core/mac/data_poll_sender.cpp

```diff
@@ -387,6 +387,15 @@ otError DataPollSender::StopFastPolls(void)
     return error;
 }
 
+void DataPollSender::ResetKeepAliveTimer(void)
+{
+    if (mTimer.IsRunning() && mPollPeriod == GetDefaultPollPeriod())
```

Comment:
Here why `GetDefaultPollPeriod()`, but not `GetKeepAlivePollPeriod()`?

Reply:
`GetDefaultPollPeriod()` is the real default keep alive period. Otherwise, skipping the poll makes the poll rate unexpected.